### PR TITLE
Identity portal

### PR DESCRIPTION
**Pull request by VictiniX888**
### Description

(_What have been done/changed_)

- Implement identity portal
- Closes HIB-147, HIB-148

### Testing plan

(_How will we test the changes requested on this PR?_)

- [ ] ...

### CODE REVIEW

File: libs/hibiscus-supabase-client/src/index.ts [hunk 1]
```diff
@@ -302,10 +378,44 @@ export class HibiscusSupabaseClient {
...
 }
 
-type UserProfileRow = Database['public']['Tables']['user_profiles']['Row'];
+export type UserProfileRow =
+  Database['public']['Tables']['user_profiles']['Row'];
 type UserProfileInsert =
   Database['public']['Tables']['user_profiles']['Insert'];
 type UserProfileUpdate =
   Database['public']['Tables']['user_profiles']['Update'];
+// type LeaderboardUpdate = Database['public']['Tables']['leaderboard']['Update'];
```
Comment (snokpok) [inline libs/hibiscus-supabase-client/src/index.ts:421]: remove comment

**Review by snokpok — CHANGES_REQUESTED** [review]
some quick fixes

File: apps/dashboard/common/add-event.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { HibiscusSupabaseClient } from '@hibiscus/hibiscus-supabase-client';
+import { Database } from '@hibiscus/types';
+import { PostgrestError } from '@supabase/supabase-js';
+import { container } from 'tsyringe';
+
+export default async function addEvent(
+  event_id: number,
+  user_id: string
+): Promise<PostgrestError | true> {
+  //Database['public']['Tables']['event_log']['Row'][]
```
[the remark this answers is not in the document]
Comment (VictiniX888) [reply apps/dashboard/common/add-event.ts]: Done!

File: apps/dashboard/common/add-event-user-id.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { HibiscusSupabaseClient } from '@hibiscus/hibiscus-supabase-client';
+import { Database } from '@hibiscus/types';
+import { PostgrestError } from '@supabase/supabase-js';
+import { container } from 'tsyringe';
+
+export default async function addEventUserId(
+  event_id: number,
+  user_id: string
+): Promise<PostgrestError | true> {
+  //Database['public']['Tables']['event_log']['Row'][]
```
[the remark this answers is not in the document]
Comment (VictiniX888) [reply apps/dashboard/common/add-event-user-id.ts]: Done!

File: apps/dashboard/common/add-event.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  // const uniqueMatchIds = new Set(
+  //   eventnameMatches.data.map((match) => match.id)
+  // );
+  // const uniqueMatches = [eventnameMatches.data];
+  // for (const match of nameMatches.data) {
+  //   if (!uniqueMatchIds.has(match.name)) {
+  //     uniqueMatchIds.add(match.name);
+  //     uniqueMatches.push(match.name);
+  //   }
+  // }
```
[the remark this answers is not in the document]
Comment (VictiniX888) [reply apps/dashboard/common/add-event.ts]: Done!

File: apps/dashboard/common/search-event-id.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { HibiscusSupabaseClient } from '@hibiscus/hibiscus-supabase-client';
+import { Database } from '@hibiscus/types';
+import { container } from 'tsyringe';
+
+export default async function searchEventId(query: number): Promise<string> {
+  //Database['public']['Tables']['events']['Row'][]
```
[the remark this answers is not in the document]
Comment (VictiniX888) [reply apps/dashboard/common/search-event-id.ts]: Done!

File: apps/dashboard/pages/identity-portal/attendee-event-scan/index.tsx [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+      if (response === true) {
+        setAlertMessage(SUCCESS_MESSAGE);
+      } else {
+        setAlertMessage(response.message);
+      }
+      setIsAlertVisible(true);
+      setResponse(null);
+      setTimeout(() => setIsAlertVisible(false), 50);
+    } else {
+      // setIsAlertVisible(false);
```
[the remark this answers is not in the document]
Comment (VictiniX888) [reply apps/dashboard/pages/identity-portal/attendee-event-scan/index.tsx]: Done!

File: libs/hibiscus-supabase-client/src/index.ts [hunk 7]
```diff
@@ -276,6 +278,80 @@ export class HibiscusSupabaseClient {
...
+  //     .from('leaderboard')
+  //     .update()
+  //     .eq('user_id', userId);
+  //     if (error) {
+  //     console.error(error);
+  //     return { message: error.message, code: error.code };
+  //   }
+
+  // }
+
```
[the remark this answers is not in the document]
Comment (VictiniX888) [reply libs/hibiscus-supabase-client/src/index.ts]: Done!

File: libs/hibiscus-supabase-client/src/index.ts (hunk 1 above)
Comment (VictiniX888) [reply libs/hibiscus-supabase-client/src/index.ts]: Done!

File: tsconfig.base.json [hunk 8]
```diff
@@ -28,7 +28,8 @@
       "@hibiscus/styles": ["libs/styles/src/index.ts"],
       "@hibiscus/types": ["libs/types/src/index.ts"],
       "@hibiscus/ui": ["libs/ui/src/index.ts"],
-      "@hibiscus/ui-kit-2023": ["libs/ui-kit-2023/src/index.ts"]
+      "@hibiscus/ui-kit-2023": ["libs/ui-kit-2023/src/index.ts"],
+      "@hibiscus/ui-portal": ["libs/ui-portal/src/index.ts"]
```
[the remark this answers is not in the document]
Comment (VictiniX888) [reply tsconfig.base.json]: Yea removed

File: package.json [hunk 9]
```diff
@@ -61,6 +61,7 @@
     "react-query": "^3.39.1",
     "react-redux": "^8.0.5",
     "react-resize-detector": "^8.0.2",
+    "react-router-dom": "^6.8.0",
```
[the remark this answers is not in the document]
Comment (VictiniX888) [reply package.json]: Yea I think we accidentally added this lol I've removed it

**Review by snokpok — APPROVED** [review]
lgtm

**Comment by VictiniX888** [comment]
Oops

**Comment by VictiniX888** [comment]
Merging

**Pull request merged**